WineCellar/WineCellar.Controller/Repositories/WineRepository.cs

```diff
@@ -49,22 +49,24 @@ public async Task<int> Update(WineRecord wine)
         return await conn.ExecuteAsync(Queries.Wine_Update, parameters, commandType: CommandType.StoredProcedure);
     }
 
-    public async Task<int> Create(WineRecord wine)
+    public async Task<int> Create(WineData wine)
     {
         DynamicParameters parameters = new();
         parameters.Add("Name", wine.Name);
-        parameters.Add("Buy", wine.Buy, dbType: DbType.Decimal, precision: 10, scale: 2);
-        parameters.Add("Sell", wine.Sell, dbType: DbType.Decimal, precision: 10, scale: 2);
-        parameters.Add("TypeId", wine.TypeId);
-        parameters.Add("CountryId", wine.CountryId);
+        parameters.Add("Buy", wine.BuyPrice, dbType: DbType.Decimal, precision: 10, scale: 2);
+        parameters.Add("Sell", wine.SellPrice, dbType: DbType.Decimal, precision: 10, scale: 2);
+        parameters.Add("CountryId", 1);
+        parameters.Add("TypeId", 1);
         parameters.Add("Picture", wine.Picture);
-        parameters.Add("Year", wine.Year);
-        parameters.Add("Content", wine.Content);
+        parameters.Add("Year", wine.Age);
+        parameters.Add("Content", wine.Contents);
         parameters.Add("Alcohol", wine.Alcohol, dbType: DbType.Decimal, precision: 10, scale: 2);
         parameters.Add("Rating", wine.Rating);
         parameters.Add("Description", wine.Description);
-
         using var conn = DataAccess.GetConnection;
-        return await conn.ExecuteScalarAsync<int>(Queries.Wine_Insert, parameters, commandType: CommandType.StoredProcedure);
+        string sql = "INSERT INTO Wine (Name, Buy, Sell, TypeId, CountryId, Picture, Year," +
```

Comment:
Waarom is dit aangepast naar een SQL string? Er zijn hiervoor Stored Procedures aangemaakt. Als de StoredProcedure niet voldoende is, laat dat weten.

Reply:
Voor de transparantie en onderhoud van het programma kan je beter geen stored procedure gebruiken, het maakt alles heel onoverzichtelijk en voegt complexiteit toe waar dat niet nodig is.

Reply:
Dit heeft niks met transparantie te maken. Onderhoud wordt alleen maar moeilijker door het niet gebruik maken van storedprocedures. In het geval een zelfde query op meerdere plekken komt, dan wordt dit onderhoud onmogelijk. Door alles in een centrale SQL project te hebben, wordt het onderhouden van queries vele malen makkelijker.

Reply:
@MonkeyOS Dat gebeurd niet want de repository zorgt er al voor dat de queries op 1 plek staan in dit geval. Stored procedures worden bijna nergens meer gebruikt. Het grote probleem wat ik met stored procedures heb is dat ik niet kan zien wat voor query er wordt uitgevoerd in de code zelf. Daarnaast kan ik een query ook niet aanpassen als dit moet. Lijkt mij niet echt handig om stored procedures te gebruiken.

Reply:
Dit is alles behalve waar, StoredProcedures worden meer en meer gebruikt. Beveiliging technisch is dit al voor veel bedrijven al genoeg reden om alleen nog maar SPs toe te laten. Daarnaast dat je "niet kan inzien wat er gebeurd" kan je wel, er is een database project waarbij de bestanden de exacte naamgeving hebben.

Overigens zijn hiervoor in de eerste weken afspraken gemaakt, ik ga je nu niet verder kunnen helpen. Zolang dit niet in het team besproken is, zal ik geen goedkeuring geven voor de MR.

Reply:
Het klopt niet dat stored procedures helpen tegen SQL injections en dergelijke. Hier zijn nog een aantal redenen voor het gebruiken van / niet gebruiken van stored procedures: https://stackoverflow.com/questions/1761601/is-the-usage-of-stored-procedures-a-bad-practice/9483781 

Als je dit graag als stored procedure wilt hebben dan kunnen we dit later nog bespreken, deze code is er nu en werkt. Dan kunnen we dit in ieder geval presenteren.